### PR TITLE
SCons: Do not enable werror=yes by default

### PR DESCRIPTION
**Pull request by akien-mga**
There are too many users who compile Godot from source and are not familiar
with the buildsystem or C/C++ compilation warnings, and thus report any kind
of yet-unfixed warning as a (often duplicate) bug.

Compiler warnings change at every compiler version and are different for each
compiler, so it's difficult to ensure that the codebase would always be 100%
warning-free, especially in the future.

I already disabled it for stable releases in #37958, but having it on non
stable commits could also become an annoyance in the future when trying to
bisect issues with a new compiler version which emits warnings unknown at
the time of commit.

TL;DR: Contributors, use `dev=yes` or `werror=yes`. CI does and won't let you
create new warnings ;)

### CODE REVIEW

**Comment by Xrayez** [comment]
> TL;DR: Contributors, use `dev=yes` or `werror=yes`. CI does and won't let you
> create new warnings ;)

You may also use [`SCONSFLAGS`](https://docs.godotengine.org/en/latest/development/compiling/introduction_to_the_buildsystem.html?highlight=sconsflags#using-the-sconsflags) to make it CI-like by default:
```
export SCONSFLAGS="werror=yes -j4"
```
